### PR TITLE
Add link checker job

### PR DESCRIPTION
**Pull request by Mixaster995**
Signed-off-by: Mikhail Avramenko <avramenkomihail15@gmail.com>

## Issue
closes https://github.com/networkservicemesh/deployments-k8s/issues/4053

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -69,3 +69,26 @@ jobs:
           BRANCH_POINT=$(git merge-base ${{github.event.pull_request.base.sha}} HEAD)
           echo "branch point: $BRANCH_POINT"
           find . -type f -not -path "./.github/*" ! -name '*.yaml' ! -name '*.yml' -exec git diff "${{ github.event.pull_request.head.sha }}".."$BRANCH_POINT" --exit-code -- {} +
+
+  linkChecker:
+    runs-on: ubuntu-latest
+    container:
+      image: lycheeverse/lychee:latest
```
Comment (denis-tingaikin) [inline .github/workflows/ci.yaml:76]: Please use the fixed version.

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -69,3 +69,26 @@ jobs:
...
+          REF=$(grep --include=\*.md -r . -e '?ref=' | head -1 | cut -d '=' -f2)
+          grep --include=\*.md -r . -e '?ref=' | while IFS= read -r line ; do
+              NEW_FILENAME=$(echo "$line" | cut -d ':' -f1)
+              if [ "$FILENAME" = "$NEW_FILENAME" ]; then
+                  continue
+              fi
+              FILENAME="$NEW_FILENAME"
+              sed -i -e "s/github.com\/networkservicemesh\/deployments-k8s/github.com\/networkservicemesh\/deployments-k8s\/tree\/$REF/g" "$FILENAME"
+              sed -i -e "s/?ref=$REF//g" "$FILENAME"
+          done
```
Comment (denis-tingaikin) [inline .github/workflows/ci.yaml:93]:
```suggestion
```

File: .github/workflows/ci.yaml [hunk 3]
```diff
@@ -69,3 +69,26 @@ jobs:
...
+          lychee -v '**/*.md' --exclude-mail
```
Comment (denis-tingaikin) [inline .github/workflows/ci.yaml:94]:
```suggestion
          lychee -v '**/*.md' --exclude-mail --exclude '.*?ref\=?.*'
```

**Review by denis-tingaikin — CHANGES_REQUESTED** [review]
Looks overcomplicated.

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (Mixaster995) [reply .github/workflows/ci.yaml]: ok, fixed

File: .github/workflows/ci.yaml (hunk 2 above)
Comment (Mixaster995) [reply .github/workflows/ci.yaml]: ok, fixed

File: .github/workflows/ci.yaml (hunk 3 above)
Comment (Mixaster995) [reply .github/workflows/ci.yaml]: ok, fixed

**Pull request merged**